packages/component-library-angular/src/checkbox/Checkbox.spec.ts

```diff
@@ -0,0 +1,365 @@
+        declarations: [UtrechtCheckboxAttr],
+      });
+      fixture.detectChanges();
+      const debugElement = fixture.debugElement.query(By.css('[type="checkbox"]input:required'));
+
+      expect(debugElement.nativeElement).toBeInTheDocument();
+    });
+  });
+
+  it('can be hidden', async () => {
+    const { fixture } = await render('<input [hidden]="true" type="checkbox" utrecht-checkbox />', {
+      declarations: [UtrechtCheckboxAttr],
+    });
+    fixture.detectChanges();
+    const debugElement = fixture.debugElement.query(By.css('[type="checkbox"]input'));
+
+    expect(debugElement.nativeElement).not.toBeVisible();
+  });
+
+  it('can have a custom class name', async () => {
+    const { fixture } = await render('<input [hidden]="true" type="checkbox" utrecht-checkbox />', {
+      declarations: [UtrechtCheckboxAttr],
+    });
+    fixture.detectChanges();
+    const debugElement = fixture.debugElement.query(By.css('[type="checkbox"]input'));
+    debugElement.nativeElement.classList.add('custom-class');
+    expect(debugElement.nativeElement).toHaveClass('custom-class');
+  });
+
+  describe('change event', () => {
+    it('can trigger a change event', async () => {
+      const { fixture } = await render(
+        '<input type="checkbox" [checked]="checked" [value]="value" utrecht-checkbox />',
+        {
+          declarations: [UtrechtCheckboxAttr],
+        },
+      );
+
+      const debugElement = fixture.debugElement.query(By.css('[type="checkbox"]input'));
+
+      fireEvent.change(debugElement.nativeElement, {
+        target: {
+          checked: 'true',
+          value: 'checked',
+        },
+      });
+      fixture.detectChanges();
+
+      expect(debugElement.nativeElement).toBeChecked();
+      expect(debugElement.nativeElement).toHaveAttribute('value', 'checked');
+    });
+
+    it('does not trigger a change event when disabled', async () => {
+      const { fixture } = await render('<input disabled="true" type="checkbox" utrecht-checkbox />', {
+        declarations: [UtrechtCheckboxAttr],
+      });
+
+      const debugElement = fixture.debugElement.query(By.css('[type="checkbox"]input'));
+      fixture.detectChanges();
+      fireEvent.change(screen.getByRole('checkbox'));
+
+      expect(debugElement.nativeElement).toBeDisabled();
+    });
+  });
+
+  describe('input event', () => {
+    it('can trigger a input event', async () => {
+      const { fixture } = await render(
+        '<input type="checkbox" [checked]="checked" [value]="value" utrecht-checkbox />',
+        {
+          declarations: [UtrechtCheckboxAttr],
+        },
+      );
+
+      const debugElement = fixture.debugElement.query(By.css('[type="checkbox"]input'));
+
+      fireEvent.input(debugElement.nativeElement, {
+        target: {
+          checked: 'true',
+          value: 'checked',
+        },
+      });
+      fixture.detectChanges();
```

Comment:
Again, would it be possible to trigger this indirectly, by simulating a `click` or `Enter` through testing library?

Reply:
Yes, of course!
I'm gonna update them!